webpack/assets/javascripts/react_app/routes/index.js

```diff
@@ -14,26 +15,27 @@ const AppSwitcher = () => (
         render={renderProps => {
           const railsContainer = document.getElementById('rails-app-content');
           if (railsContainer) railsContainer.remove();
-          currentLocation = renderProps.location;
+          currentPath = renderProps.location.pathname;
 
           return render(renderProps);
         }}
       />
     ))}
     <Route
-      render={child => {
-        if (
-          currentLocation &&
-          currentLocation.pathname !== child.location.pathname
-        ) {
-          const useTurbolinks =
-            child.location.state &&
-            child.location.state.useTurbolinks &&
-            !window.history.state.turbolinks; // visit() already called
+      render={({ location }) => {
+        // the routeProps.location.pathname doesn't give you the full URI path.
+        const uri = new URI();
+        const nextPath = uri.pathname();
 
-          if (useTurbolinks) window.Turbolinks.visit(child.location.pathname);
+        if (currentPath !== nextPath) {
+          currentPath = nextPath;
+          const { state: { useTurbolinks } = {} } = location;
+          const turbolinksVisitCalled = !window.history.state.turbolinks;
```

Comment:
can you please use the history prop from the `renderProprs` instead of the global one?

Reply:
tried that but the history object is different and doesn't contain the `turbolinks` object for some reason.. :(